IntroToCSharp/Pages/Lessons/Collections/Arrays.razor

```diff
@@ -0,0 +1,379 @@
+@inherits CaptainCoder.Page
+@page "/lessons/collections/arrays"
+
+<Title>Arrays</Title>
+
+
+<Section Title="Introdution">
+    <MudText>
+        An array is a data structure that stores a pre-defined number of variables of the same data type within its bounds.
+         The important part of this definition to take note of is that an array is predefined, meaning that you have to know 
+         how many variables are going to be within the array before you create it.
+         This is because arrays are immutable data structures which means after they are created their <b>length</b> can't be changed.
```

Comment:
A nice start. What do you think about:

```
Often times when programming we find that we need to track multiple related values of the same type. For example, you might be writing a game in which you have multiple enemies in a scene. In this case, you might write the code:

string enemy0 = "Goblin";
string enemy1 = "Rat";
string enemy2 = "Skeleton";

For just a few enemies, this might seem reasonable. However, as the number of enemies increases, this can become unwieldy.

Enter the array data structure:

An array is a reference data type that allows us to store multiple values of the same type in a single variable.
```

I would move the part about defining the size/immutability into a section later. 
Also, it is important to note that arrays are mutable but specifically their size is immutable.

Reply:
I like it.  I will add a more introductory section like you suggested. That feels like a cleaner introduction.